src/presentation/dtos/device-dto.ts

```diff
@@ -1,25 +1,25 @@
-import {INewsDevice, TelemetricsDevice} from '../../model/entities/device'
-import {DeviceType} from '../../model/enums/device-type'
-import {StatusCode} from '../../model/enums/status-code'
+import { CoreDevice, Device, INewsDevice, TelemetricsDevice } from '../../model/entities/device'
+import { DeviceType } from '../../model/enums/device-type'
+import { StatusCode } from '../../model/enums/status-code'
 
-export interface DeviceDtoInterface {
-  type: DeviceType
-  id: string
-  name: string
-  isConnected: boolean
-  statusCode: StatusCode
-  statusMessage: string
-}
+// export interface DeviceDtoInterface {
+//   type: DeviceType
+//   id: string
+//   name: string
+//   isConnected: boolean
+//   statusCode: StatusCode
+//   statusMessage: string
+// }
 
-export abstract class DeviceDto implements DeviceDtoInterface {
+export abstract class DeviceDto implements CoreDevice {
```

Comment:
These Dtos doesn't need to implement any interface. You are declaring all the variables in the dto anyway, so you gain nothing from the interface.

Also, you are implementing `CoreDevice`. `CoreDevice` is only used in the flow that gives us statusMessages from the Devices that Core manages. It should not be used anywhere else.